build.gradle

```diff
@@ -499,6 +502,32 @@ subprojects {
         maxFailures = userMaxTestRetryFailures
       }
     }
+    
+    // As we process results, check if there were any test failures.
+    afterSuite { desc, result ->
+      if (result.resultType == TestResult.ResultType.FAILURE) {
```

Comment:
not sure whether it works with `maxTestRetries` and `maxTestRetryFailures`? It should show "successful" if the failed test pass after retry.

maybe we can use `finalizedBy` instead. for example:
```
  tasks.test {
    finalizedBy("copyJunitXml")
  }
  tasks.register("copyJunitXml") {
    doLast {
      def dest = rootProject.layout.buildDirectory.dir("junit-xml/${project.name}").get().asFile
      println "Copy JUnit XML for ${project.name} to $dest"
      ant.copy(todir: "$dest") {
        ant.fileset(dir: "${test.reports.junitXml.entryPoint}")
      }
    }
  }
```

Reply:
I did try `finalizedBy`, unfortunately it is always run even if a task was UP-TO-DATE or FROM-CACHE. This results in all the XML files being copied, even if they were from the cache.

Back here https://github.com/apache/kafka/pull/17066#discussion_r1744071589 you suggested using the GITHUB_ACTIONS env as a way to control caching. I think we can do something similar here.

If GITHUB_ACTIONS is set, we'll always set `ignoreFailures=true`. This ensures our `doLast` is called and the files are copied. Since the CI pipeline relies on the downstream "Parse JUnit Tests" job for detecting failures, we don't care if Gradle exits with 1 or 0.

Reply:
> If GITHUB_ACTIONS is set, we'll always set ignoreFailures=true. This ensures our doLast is called and the files are copied. Since the CI pipeline relies on the downstream "Parse JUnit Tests" job for detecting failures, we don't care if Gradle exits with 1 or 0.

that is a better approach :)